dgraphtest/image.go

```diff
@@ -16,6 +16,107 @@
 
 package dgraphtest
 
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/pkg/errors"
+)
+
 func (c *LocalCluster) dgraphImage() string {
 	return "dgraph/dgraph:local"
 }
+
+func (c *LocalCluster) setupBinary() error {
+	ctx, cancel := context.WithTimeout(context.Background(), cloneTimeout)
+	defer cancel()
+	repo, err := git.PlainOpen(repoDir)
+	if err != nil && err == git.ErrRepositoryNotExists {
+		c.log("cloning repo")
+		repo, err = git.PlainCloneContext(ctx, repoDir, false, &git.CloneOptions{
+			URL: dgraphRepoUrl,
+		})
+		if err != nil {
+			return errors.Wrap(err, "error while checking out dgraph git repo")
+		}
+	} else if err != nil {
+		return errors.Wrap(err, "error while opening git repo")
+	} else {
+		if err := repo.Fetch(&git.FetchOptions{}); err != nil && err != git.NoErrAlreadyUpToDate {
+			return errors.Wrap(err, "error while fetching git repo")
+		}
+	}
+	hash, err := repo.ResolveRevision(plumbing.Revision(c.conf.version))
+	if err != nil {
+		return err
+	}
+	if err := checkoutGitRepo(repo, hash); err != nil {
+		return err
+	}
+	// make dgraph of specific version
+	c.log("building dgraph binary")
+	if err := buildDgraphBinary(repoDir, binDir, c.conf.version); err != nil {
+		return err
+	}
+	if err := copy(filepath.Join(binDir, fmt.Sprintf(binaryName, c.conf.version)),
```

Comment:
we are using the copy function everywhere to copy the binary from binDir to c.tempBinDir. Why don't we just modify copy function to something like -
```
func copyBinary(fromDir, toDir, version string) error {}
```

Reply:
Yeah, We can do that but if we  need  a normal copy function  in future then better have a function which can do both things.

Reply:
yeah, I think that is fine. We can keep the copy function, but move this big line into a small function